garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -3,10 +3,257 @@ local meta = FindMetaTable( "Vector" )
+function meta:GetRotated(...)
+	local v = Vector(self)
+	v:Rotate(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a multiplied vector
+-----------------------------------------------------------]]
+function meta:GetMul(...)
+	local v = Vector(self)
+	v:Mul(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of an added vector
+-----------------------------------------------------------]]
+function meta:GetAdd(...)
+	local v = Vector(self)
+	v:Add(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a vector
+-----------------------------------------------------------]]
+function meta:AddUnpacked(x, y, z)
+	local vx, vy, vz = self:Unpack()
+	self:AddUnpacked(vx + x, vy + y, vz + z)
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a copy vector
+-----------------------------------------------------------]]
+function meta:GetAddUnpacked(...)
+	local v = Vector(self)
+	v:AddUnpacked(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a subracted vector
+-----------------------------------------------------------]]
+function meta:GetSub(...)
+	local v = Vector(self)
+	v:Sub(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Subtracts a unpacked x, y and z from a vector
+-----------------------------------------------------------]]
+function meta:SubUnpacked(x, y, z)
+	local vx, vy, vz = self:Unpack()
+	self:SubUnpacked(vx - x, vy - y, vz - z)
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a copy vector
+-----------------------------------------------------------]]
+function meta:GetSubUnpacked(...)
+	local v = Vector(self)
+	v:SubUnpacked(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a divided vector
+-----------------------------------------------------------]]
+function meta:GetDiv(...)
+	local v = Vector(self)
+	v:Div(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Converts the vector to bisector of two vectors
+Converts vector to result of a*|b| + b*|a|
+-----------------------------------------------------------]]
+function meta:Bisect(vec)
+	local na, nb = self:Length(), vec:Length()
+	self:Mul(nb)
+	self:Add(vec:GetMul(na))
+end
+
+--[[---------------------------------------------------------
+Returns a copy of bisector of two vectors
+-----------------------------------------------------------]]
+function meta:GetBisected(...)
+	local v = Vector(self)
+	v:Bisect(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Convets the vector to negated version of itself
+-----------------------------------------------------------]]
+function meta:Negate()

```

Comment:
This should be removed for an engine implementation - I would go ahead and also remove GetNegated until the bindings are made so this can be used on people's servers in the meantime.

Reply:
Removed for you to do engine implementation.